src/Monolog/Handler/ElasticsearchHandler.php

```diff
@@ -54,11 +56,15 @@ class ElasticsearchHandler extends AbstractProcessingHandler
     protected $options = [];
 
     /**
-     * @param Client  $client  Elasticsearch Client object
-     * @param mixed[] $options Handler configuration
+     * @param Client|Client8 $client  Elasticsearch Client object
+     * @param mixed[]        $options Handler configuration
      */
-    public function __construct(Client $client, array $options = [], $level = Logger::DEBUG, bool $bubble = true)
+    public function __construct($client, array $options = [], $level = Logger::DEBUG, bool $bubble = true)
     {
+        if (!$client instanceof Client && !$client instanceof Client8) {
+            throw new \InvalidArgumentException('Elasticsearch\Client or Elastic\Elasticsearch\Client instance required');
```

Comment:
I would throw a TypeError here, so that it is the same type than when using an argument type